--- openrc-0.11.6/runlevels/Makefile	2012-11-30 19:07:33.000000000 +0100
+++ openrc-0.11.6/runlevels/Makefile.new	2012-12-05 02:03:09.284176658 +0100
@@ -53,34 +53,22 @@
 	if ! test -d "${SYSINITDIR}"; then \
 		${INSTALL} -d ${SYSINITDIR} || exit $$?; \
 		for x in ${SYSINIT}; do \
-			if test -n "${PREFIX}"; then \
-				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
-			fi; \
 			ln -snf ${INITDIR}/"$$x" ${SYSINITDIR}/"$$x" || exit $$?; done \
 	fi
 	if ! test -d "${BOOTDIR}"; then \
 		${INSTALL} -d ${BOOTDIR} || exit $$?; \
 		for x in ${BOOT}; do \
-			if test -n "${PREFIX}"; then \
-				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
-			fi; \
 			ln -snf ${INITDIR}/"$$x" ${BOOTDIR}/"$$x" || exit $$?; \
 		done \
 	fi
 	if ! test -d "${DEFAULTDIR}"; then \
 		${INSTALL} -d ${DEFAULTDIR} || exit $$?; \
 		for x in ${DEFAULT}; do \
-			if test -n "${PREFIX}"; then \
-				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
-			fi; \
 			ln -snf ${INITDIR}/"$$x" ${DEFAULTDIR}/"$$x" || exit $$?; done \
 	fi
 	if ! test -d "${SHUTDOWNDIR}"; then \
 		${INSTALL} -d ${SHUTDOWNDIR} || exit $$?; \
 		for x in ${SHUTDOWN}; do \
-			if test -n "${PREFIX}"; then \
-				grep -q "keyword .*-prefix" ${INITFILES}/"$$x" && continue; \
-			fi; \
 			ln -snf ${INITDIR}/"$$x" ${SHUTDOWNDIR}/"$$x" || exit $$?; done \
 	fi
 
